packages/next/next-server/server/next-server.ts

```diff
@@ -409,7 +409,7 @@ export default class Server {
                   destinationCompiler = pathToRegexp.compile(
                     followingRoute.destination
                   )
-                  newUrl = destinationCompiler(newParams)
+                  newUrl = destinationCompiler(curParams)
```

Comment:
This was a bug when doing https://github.com/zeit/next.js/pull/9435/files#diff-1d261dc0b5457a82dd7b972d5fe8bbf8R42-R51 it would fail